### PR TITLE
ONCE dataset

### PR DESCRIPTION
**Pull request by mariarzv**

### CODE REVIEW

File: patch_scene.py [hunk 1]
```diff
@@ -111,8 +112,9 @@ def __patch_scene(scene_id: str,
 
 def parse_arguments():
     parser = argparse.ArgumentParser(description="patch scene arguments")
-    parser.add_argument("--dataset", type=str, choices=['nuscenes', 'waymo'], default='nuscenes', help="Dataset.")
+    parser.add_argument("--dataset", type=str, choices=['nuscenes', 'once', 'waymo'], default='nuscenes', help="Dataset.")
     parser.add_argument("--version", type=str, default="v1.0-mini", help="NuScenes version.")
+    parser.add_argument("--type", type=str, choices=['train', 'test', 'val', 'raw_small', 'raw_medium', 'raw_large'], default="train", help="Once dataset type.")
```
Comment (st235) [inline patch_scene.py]: [Optional] We can call it `split`

File: src/accumulation/default_accumulator_strategy.py [hunk 2]
```diff
@@ -14,4 +14,10 @@ def on_merge(self,
                  initial_point_cloud: np.ndarray,
                  next_point_cloud: np.ndarray,
                  frame_no: int) -> np.ndarray:
-        return np.concatenate((initial_point_cloud, next_point_cloud), axis=1)
+        if next_point_cloud.size == 0:
+            return initial_point_cloud
+        else:
```
Comment (st235) [inline src/accumulation/default_accumulator_strategy.py:21]: Can we change it to something like 

```
elif
else
```

File: src/datasets/once/once_dataset.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    def get_scene_iterator(self, scene_id: str) -> Dataset.SceneIterator:
+        assert scene_id in self.__scenes_lookup
+        # scene_descriptor = self.__load_scene_descriptor(scene_id=scene_id)
+        return OnceSceneIterator(scene=self.__scenes_lookup[scene_id],
+                                 once=self.__once)
+
+    def load_frame_patcher(self, frame_id: str,
+                                 scene_id: str) -> FramePatcher:
+        # scene_descriptor = self.__load_scene_descriptor(scene_id=scene_id)
```
Comment (st235) [inline src/datasets/once/once_dataset.py]: let's remove this

File: src/datasets/once/once_frame_patcher.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    def __init__(self,
+                 seq_id: str,
+                 frame_id: str,
+                 frame_point_cloud: np.ndarray,
+                 # frame_descriptor: dict,
+                 once: ONCE):
+        self.__seq_id = seq_id
+        self.__frame_id = frame_id
+        self.__frame_point_cloud = frame_point_cloud
+        # self.__frame_descriptor = frame_descriptor
```
Comment (st235) [inline src/datasets/once/once_frame_patcher.py]: Can you, please, remove the commented lines here and in other places in the codebase?

Comment (mariarzv) [reply src/datasets/once/once_frame_patcher.py]: removed

File: src/datasets/once/once_dataset.py (hunk 3 above)
Comment (mariarzv) [reply src/datasets/once/once_dataset.py]: removed everywhere in once

File: patch_scene.py (hunk 1 above)
Comment (mariarzv) [reply patch_scene.py]: renamed

File: src/accumulation/default_accumulator_strategy.py (hunk 2 above)
Comment (mariarzv) [reply src/accumulation/default_accumulator_strategy.py:21]: fixed

File: src/datasets/once/once_tracking.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+
+def parallel_process(dataset,
+                     dataset_root,
+                     scene,
+                     pickle_data,
+                     sequences_to_frames_lookup,
+                     frame_id_to_annotations_lookup,
+                     outfile):
+    num_workers = 2  # multiprocessing.cpu_count()
```
Comment (st235) [inline src/datasets/once/once_tracking.py]: Let's uncomment this change

**Pull request merged**